ihmc-perception/src/main/java/us/ihmc/perception/ImageSensorPublishThread.java

```diff
@@ -6,13 +6,17 @@
 import us.ihmc.ros2.ROS2Topic;
 import us.ihmc.sensors.ImageSensor;
 
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Map.Entry;
 
 public class ImageSensorPublishThread extends RepeatingTaskThread
 {
    private final Map<Integer, ROS2Topic<? extends Packet<?>>> imageKeyToTopicMap;
+   private final Map<Integer, Long> lastPublishedSequenceNumbers = new HashMap<>();
    private final RawImagePublisher publisher;
+   private boolean publishingIsThrottled = false;
```

Comment:
This tells the thread whether to use the sensor's frequency, or the frequency set by the programmer

Reply:
Would it make more sense to get the sensor's frequency direction from the sensor? and then use that as the `ImageSensorPublishThread` frequency?

Reply:
I don't really like that idea because it can introduce unnecessary latency. Ideally we publish as soon as the sensor gets a new image. That's why the 'waitForGrab()' method is good. But if you want to publish at an arbitrary frequency, you can't use `waitForGrab()` so I just accept that you'll have some added latency.